core/src/main/java/net/opentsdb/query/execution/serdes/JsonV3QuerySerdes.java

```diff
@@ -226,12 +239,12 @@ public Object call(final ArrayList<TimeSeriesStringId> ids)
 
           json.writeArrayFieldStart("data");
           int idx = 0;
-
-          boolean wasStatus = false;
-          boolean wasEvent = false;
-          String namespace = null;
-          if (opts.getParallelThreshold() > 0 &&
-              result.timeSeries().size() > opts.getParallelThreshold()) {
+          AtomicInteger ai = new AtomicInteger(0);
+          AtomicBoolean wasStatus = new AtomicBoolean(false);
+          StringBuilder namespace = new StringBuilder();
+          Collection<QueryNode> topNExists = result.source().pipelineContext().downstreamOfType(result.source(), TopN.class);
+          if (topNExists == null || topNExists.size() == 0) {
```

Comment:
And need threshold limit so we only do this if we have more than X number of time series..

Reply:
We can always run it in parallel mode even if it's just a single timeseries